### PR TITLE
Task 1

### PR DESCRIPTION
**Pull request by GarinaKristina**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,15 +9,20 @@
...
     "allure-commandline": "^2.21.0",
+    "babel-eslint": "^10.1.0",
     "chromedriver": "^110.0.0",
+    "eslint": "^8.35.0",
     "rimraf": "^4.1.2",
     "wdio-chromedriver-service": "^8.1.1"
   },
   "scripts": {
-    "prewdio": "rimraf allure-results",
+    "prewdio": "rimraf allure-results && npx eslint ./pageobjects/*.js",
```
Comment (jkalandarov) [inline package.json]: `npx eslint ./pageobjects/**/*.js`

File: pageobjects/MainPage.js [hunk 2]
```diff
@@ -16,9 +16,14 @@ class MainPage extends BasePage {
       buttonName: 'Submit',
       buttonLocator: '//button[@type="submit"] | //*[class="sb-searchbox__button"]',
     },
+    CalendarButton: {
+      buttonName: 'Calendar',
+      buttonLocator: '.ed2ff9f661',
+    },
```
Comment (jkalandarov) [inline pageobjects/MainPage.js]: `get ['Calendar button']() {
   return '.ed2ff9f661';
}`

File: pageobjects/MainPage.js [hunk 3]
```diff
@@ -16,9 +16,14 @@ class MainPage extends BasePage {
...
   };
 
   firstLocation = (locationName) => `//*[normalize-space()="${locationName}"]`;
```
Comment (jkalandarov) [inline pageobjects/MainPage.js]: The name of the function should be a verb, example: getFirstLocation(), findFirstLocation(), etc.

File: pageobjects/MainPage.js [hunk 4]
```diff
@@ -16,9 +16,14 @@ class MainPage extends BasePage {
...
+  dataInCalendar = (data) => `//*[@aria-label="${data}"]`;
```
Comment (jkalandarov) [inline pageobjects/MainPage.js]: Change function name to a verb

File: pageobjects/MainPage.js [hunk 5]
```diff
@@ -53,16 +58,26 @@ class MainPage extends BasePage {
...
+    await ElementAction.sendKeys('\uE007');
   }
 
   async titleContains(title) {
     await expect(browser).toHaveTitleContaining(title);
   }
-}
 
+  async chooseTheDataFromCalendar(dataOne, dataTwo) {
+    const dataFrom = this.dataInCalendar(dataOne);
```
Comment (jkalandarov) [inline pageobjects/MainPage.js]: change parameters' names

File: pageobjects/MainPage.js [hunk 6]
```diff
@@ -53,16 +58,26 @@ class MainPage extends BasePage {
 
   async fillTheField(fieldName, text) {
     const element = this.getFieldByName(fieldName);
-    console.log(element);
     await Waiters.waitElementIsDisplayed(element);
     await ElementAction.setValueInField(element, text);
     await ElementAction.click(this.firstLocation(text));
+    await ElementAction.click(element);
+    await ElementAction.sendKeys('\uE007');
   }
```
Comment (jkalandarov) [inline pageobjects/MainPage.js]: create move '\uE007' to constants

File: pageobjects/MainPage.js (hunk 5 above)
Comment (jkalandarov) [reply pageobjects/MainPage.js]: change function name to `selectDatesFromCalender`

File: wdio.conf.js [hunk 7]
```diff
@@ -46,7 +46,7 @@ export const config = {
     ],
   ],
   cucumberOpts: {
-    require: ['./features/step-definitions/steps.js'],
+    require: ['./step-definitions/*.js'],
```
Comment (jkalandarov) [inline wdio.conf.js]: `./step-definitions/**/*.js`

**Review by jkalandarov — CHANGES_REQUESTED** [review]
Please make changes

File: package.json (hunk 1 above)
Comment (GarinaKristina) [reply package.json]: Done

File: wdio.conf.js (hunk 7 above)
Comment (GarinaKristina) [reply wdio.conf.js]: Done

File: features/OpenWebSite.feature [hunk 8]
```diff
@@ -3,5 +3,5 @@ Feature: Posobolity open the website
   Scenario: As a user, I can open the website and see the main page
 
     Given Open www.booking.com website
-    When Click on "Accept cookie" button
+    When Click on "Main Page > Accept cookie button"
```
Comment (jkalandarov) [inline features/OpenWebSite.feature]: Are you going to parse the string?

File: src/pageobjects/BasePage.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+import { expect as expectChai } from 'chai';
+
+export default class BasePage {
+  open() {
+    return browser.url(config.baseUrl);
+  }
+
+  static get enterKey() {
+    return '\uE007';
+  }
```
Comment (jkalandarov) [inline src/pageobjects/BasePage.js]: Are you going to use this static method only within the class?

File: src/pageobjects/MainPage.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  get ['Accept cookie button']() {
+    return '#onetrust-accept-btn-handler';
+  }
+
+  travelStartDateButton = 'button[data-testid="date-display-field-start"]';
+  submitButton = '//button[@type="submit"] | //*[class="sb-searchbox__button"]';
+  calendarButton = '.ed2ff9f661';
+  searchBoxField = '//input[@name="ss"]';
+  getFirstLocation = (locationName) => `//*[normalize-space()="${locationName}"]`;
+  getDataInCalendar = (data) => `//*[@aria-label="${data}"]`;
```
Comment (jkalandarov) [inline src/pageobjects/MainPage.js]: Transform the to getters.

File: src/pageobjects/MainPage.js [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+  async fillTheField(fieldName, text) {
```
Comment (jkalandarov) [inline src/pageobjects/MainPage.js]: I recommend renaming the method. Something like `setInputValue`

File: src/pageobjects/MainPage.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    await Waiters.waitElementIsDisplayed(element);
+    await ElementAction.setValueInField(element, text);
+    await ElementAction.click(this.getFirstLocation(text));
+    await ElementAction.click(element);
+    await ElementAction.sendKeys(BasePage.enterKey);
+  }
+
+  async selectTheDataFromCalendar(dataOne, dataTwo) {
```
Comment (jkalandarov) [inline src/pageobjects/MainPage.js]: Please remove `The` from the method name, and rename parameters' names: `startDate, endDate`.

**Review by jkalandarov — CHANGES_REQUESTED** [review]
Please make changes according to the comments.

File: pageobjects/MainPage.js (hunk 2 above)
Comment (GarinaKristina) [reply pageobjects/MainPage.js]: Done

File: pageobjects/MainPage.js (hunk 3 above)
Comment (GarinaKristina) [reply pageobjects/MainPage.js]: chooseLocation

File: pageobjects/MainPage.js (hunk 4 above)
Comment (GarinaKristina) [reply pageobjects/MainPage.js]: I removed this code, YAGNI

File: pageobjects/MainPage.js (hunk 5 above)
Comment (GarinaKristina) [reply pageobjects/MainPage.js]: I removed this code too

File: pageobjects/MainPage.js (hunk 6 above)
Comment (GarinaKristina) [reply pageobjects/MainPage.js]: Done

File: features/OpenWebSite.feature (hunk 8 above)
Comment (GarinaKristina) [reply features/OpenWebSite.feature]: image
I don`t know why you see the old code

File: src/helpers/ElementAction.js [hunk 13]
```diff
@@ -19,5 +19,15 @@ class ElementAction {
...
     }
   }
+  async sendKeys(keyText) {
+    try {
+      await browser.keys(keyText);
+    } catch (error) {
+      throw ErrorWrapper.elementError(error, keyText);
+    }
+  }
+  async fillField(selector, text) {
```
Comment (jkalandarov) [inline src/helpers/ElementAction.js]: you don't need this method

File: src/pageobjects/BasePage.js (hunk 9 above)
Comment (GarinaKristina) [reply src/pageobjects/BasePage.js]: Removed static

File: src/pageobjects/MainPage.js (hunk 10 above)
Comment (GarinaKristina) [reply src/pageobjects/MainPage.js]: This is also removed, because YAGNI

File: src/pageobjects/MainPage.js (hunk 11 above)
Comment (GarinaKristina) [reply src/pageobjects/MainPage.js]: Done

**Comment by jkalandarov** [comment]
put report folder in gitignore

File: src/pageobjects/MainPage.js (hunk 12 above)
Comment (GarinaKristina) [reply src/pageobjects/MainPage.js]: I delete this code, because YAGNI

File: src/helpers/ElementAction.js (hunk 13 above)
Comment (GarinaKristina) [reply src/helpers/ElementAction.js]: image
image
image

**Comment by GarinaKristina** [comment]
HI, I believe , I fixed all comments :)

**Comment by jkalandarov** [comment]
delete allure-report folder, commit and push

File: src/helpers/GetData.js [hunk 14]
```diff
@@ -9,6 +9,7 @@ class GetData {
       await element.scrollIntoView({ block: 'center', inline: 'center' });
       if (visible) {
         await Waiters.waitElementIsDisplayed(elementLocator);
+        var a;
```
Comment (jkalandarov) [inline src/helpers/GetData.js]: what is `var a`?

Comment (GarinaKristina) [reply src/helpers/GetData.js]: Oh. yes. I am testing eslint rules and forgot about it.Thanks

**Pull request merged**